osquery/tables/system/windows/handles.cpp

```diff
@@ -0,0 +1,341 @@
+}
+
+std::string getHandleAttributes(const ULONG &handleAttributes) {
+
+  std::stringstream ss;
+
+  if ((handleAttributes & OBJ_INHERIT) == OBJ_INHERIT) {
+    ss << "OBJ_INHERIT";
+  }
+  if ((handleAttributes & OBJ_PERMANENT) == OBJ_PERMANENT) {
+    ss << ",OBJ_PERMANENT";
+  }
+  if ((handleAttributes & OBJ_EXCLUSIVE) == OBJ_EXCLUSIVE) {
+    ss << ",OBJ_EXCLUSIVE";
+  }
+  if ((handleAttributes & OBJ_CASE_INSENSITIVE) == OBJ_CASE_INSENSITIVE) {
+    ss << ",OBJ_CASE_INSENSITIVE";
+  }
+  if ((handleAttributes & OBJ_OPENIF) == OBJ_OPENIF) {
+    ss << ",OBJ_OPENIF";
+  }
+  if ((handleAttributes & OBJ_OPENLINK) == OBJ_OPENLINK) {
+    ss << ",OBJ_OPENLINK";
+  }
+  if ((handleAttributes & OBJ_KERNEL_HANDLE) == OBJ_KERNEL_HANDLE) {
+    ss << ",OBJ_KERNEL_HANDLE";
+  }
+  if ((handleAttributes & OBJ_FORCE_ACCESS_CHECK) == OBJ_FORCE_ACCESS_CHECK) {
+    ss << ",OBJ_FORCE_ACCESS_CHECK";
+  }
+  if ((handleAttributes & OBJ_IGNORE_IMPERSONATED_DEVICEMAP) ==
+      OBJ_IGNORE_IMPERSONATED_DEVICEMAP) {
+    ss << ",OBJ_IGNORE_IMPERSONATED_DEVICEMAP";
+  }
+  if ((handleAttributes & OBJ_DONT_REPARSE) == OBJ_DONT_REPARSE) {
+    ss << ",OBJ_DONT_REPARSE";
+  }
+  if ((handleAttributes & OBJ_VALID_ATTRIBUTES) == OBJ_VALID_ATTRIBUTES) {
+    ss << ",OBJ_VALID_ATTRIBUTES";
+  }
+
+  std::string handlesAttrsString = ss.str();
+  if (!handlesAttrsString.empty() && handlesAttrsString.front() == ',') {
+    handlesAttrsString.erase(handlesAttrsString.begin());
+  }
+  return handlesAttrsString;
+}
+
+QueryData genHandles(QueryContext &context) {
+  QueryData rows;
+  ULONG i;
+  // DWORD   currentPid;
+  HANDLE processHandle;
+  HANDLE processDupHandle;
+  PSYSTEM_HANDLE_INFORMATION_EX handleInfo;
+  SYSTEM_HANDLE_TABLE_ENTRY_INFO_EX handle;
+  HMODULE ntdllModule = nullptr;
+
+  GetModuleHandleExA(GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT, "ntdll.dll",
+                     &ntdllModule);
+  auto _NtDuplicateObject = reinterpret_cast<NtDuplicateObject>(
+      GetProcAddress(ntdllModule, "NtDuplicateObject"));
+  auto _NtQueryObject = reinterpret_cast<NtQueryObject>(
+      GetProcAddress(ntdllModule, "NtQueryObject"));
+
+  if ((handleInfo = (PSYSTEM_HANDLE_INFORMATION_EX)malloc(
+           BUFF_SIZE)) == NULL) {
+    VLOG(1) << L"Could not allocate memory for handleInfo";
+    return rows;
+  }
+  auto const guard_handle_info = scope_guard::create(
+      [handleInfo]() { free(handleInfo); });
+
+  auto status = getSystemHandles(handleInfo);
+  auto const guard_process_dup_handle = scope_guard::create(
+      [processDupHandle]() { CloseHandle(processDupHandle); });
+
+  if (!status.ok()) {
+    VLOG(1) << L"Unable to get system handles: " << status.getCode();
+    return rows;
+  }
+
+  // currentPid = GetCurrentProcessId();
+  for (i = 0; i < handleInfo->NumberOfHandles; i++) {
+    // if (handleInfo->Handles[i].UniqueProcessId == currentPid) {
+    //     // Do not get handles for the current process
+    //     continue;
+    // }
```

Comment:
Remove commented code. (And why would we omit handles for the current process)

Reply:
done. the table now displays handles for the current process